### ISSUE TITLE
Connecting to a cluster that has a different broadcast address from RPC address seems to break

### ISSUE THREAD

**Issue by pcn** [Netflix/astyanax#452]
TL;DR:  Is there any reason why astyanax uses the endpoint rather than the rpc endpoint?

I have a cluster in EC2 that is using a broadcast_address that is the public address, and an RPC address that is the private address.

When astyanax tries to connect, it does a TokenRange.getEndpoint_details() call, and that only provide the endpoints, not the rpc_endpoints.

How does this get managed at netflix when you have multi-region clusters?  The ec2 multi-region snitch is made to support this configuration, but it seems like astyanax doesn't?

As an option, we're trying this:

https://github.com/dsiegel/astyanax/commit/b2b8b6af56f46d8d122d5d8c25fcf753f962df63

but it looks like that has its own caveats WRT not guaranteeing the order of the endpoint and the rpc_endpoints.  It seems like the public methods StorageService.describeLocalRing and describeRing(String keyspace) provide a mapping with both the rpc and endpoint info.
